LaTeX/LaTeX.sublime-syntax

```diff
@@ -110,67 +118,120 @@ contexts:
         - include: general-commands
         - include: macro-braces
-    - match: |-
-        (?x)
-        (
-          (\\)
-          DeclareMathOperator\*?
-        )
-        (?:
-          (\{)(\\[A-Za-z@]+)(\})
-          | (\\[A-Za-z@])+
-        )
-        (?:(\[)(?:[^\]]*)(\]))?
-        (\{)
+
+    - include: else-pop
+
+  newcommand-optarg:
+    - include: paragraph-pop
+    - match: '\['
+      scope: punctuation.definition.group.bracket.begin.latex
+      push:
+        - meta_scope: meta.function.parameters.default-value.latex
+        - match: '\]'
+          scope: punctuation.definition.group.bracket.end.latex
+          pop: 2
+        - include: general-constants
+        - include: general-commands
+        - include: macro-braces
+    - include: else-pop
+
+  newcommand-commandname:
+    - include: paragraph-pop
+    # version one: brace-wrapped command name
+    - match: '(\{)\s*((\\)[A-Za-z@]+)\s*(\})'
+      scope: meta.function.latex
       captures:
-        1: support.function.declare-math-operator.latex storage.modifier.declare-math-operator.latex
-        2: punctuation.definition.backslash.latex
-        3: punctuation.definition.group.brace.begin.latex
-        4: support.function.latex entity.name.declare-math-operator.latex
-        5: punctuation.definition.group.brace.end.latex
-        6: support.function.latex
-        7: punctuation.definition.group.bracket.begin.latex
-        8: punctuation.definition.group.bracket.end.latex
-        9: punctuation.definition.group.brace.begin.latex
+        1: punctuation.definition.group.brace.begin.latex
+        2: entity.name.newcommand.latex
+        3: punctuation.definition.backslash.latex
+        4: punctuation.definition.group.brace.end.latex
+      pop: true
+    # version two: directly written
+    - match: '(\\)[A-Za-z@]+'
+      scope: meta.function.latex entity.name.newcommand.latex
+      captures:
+        1: punctuation.definition.backslash.latex
+      pop: true
+    - include: else-pop
+
+
+  # we need a separate context for command definitions, because they can contain
+  # un-balanced elements.
+  newcommand-definition:
+    - include: paragraph-pop
+    # usually, a command definition is a brace-enclosed group of tokens
+    - match: '\{'
+      scope: punctuation.definition.group.brace.begin.latex
       push:
-        - meta_scope: meta.function.declare-math-operator.latex
+        - meta_scope: meta.function.definition.latex
         - match: '\}'
           scope: punctuation.definition.group.brace.end.latex
-          pop: true
+          pop: 2
         - include: general-constants
         - include: general-commands
         - include: macro-braces
 
+    # but in principle, a single token is also valid,
+    # thus either a command sequence ...
+    - match: '(\\)[A-Za-z@]+'
+      captures:
+        1: punctuation.definition.backslash.latex
+      scope: meta.function.definition.latex support.function.general.latex
```

Comment:
The `scope` should always be the first before `captures` or other keys follow. Just to have a common scheme of how rules are organized.

```suggestion
      scope: meta.function.definition.latex support.function.general.latex
      captures:
        1: punctuation.definition.backslash.latex
```